service/src/test/java/org/cbioportal/service/impl/StudyViewServiceImplTest.java

```diff
@@ -255,6 +256,79 @@ public void getCNAAlterationCountByGenes() throws Exception {
+        genomicDataFilters.add(genomicDataFilter1);
+        genomicDataFilters.add(genomicDataFilter2);
+
+        List<MolecularProfile> molecularProfiles = new ArrayList<>();
+        MolecularProfile molecularProfile = new MolecularProfile();
+        molecularProfile.setCancerStudyIdentifier(BaseServiceImplTest.STUDY_ID);
+        molecularProfile.setStableId(BaseServiceImplTest.STUDY_ID + "_" + BaseServiceImplTest.MOLECULAR_PROFILE_ID_A);
+        molecularProfiles.add(molecularProfile);
+        
+        Mockito.when(molecularProfileService.getMolecularProfilesInStudies(studyIds, "SUMMARY"))
+            .thenReturn(molecularProfiles);
+        
+        Map<String, List<MolecularProfile>> molecularProfileMap = new HashMap<>();
+        molecularProfileMap.put(BaseServiceImplTest.PROFILE_TYPE_1, molecularProfiles);
+        molecularProfileMap.put(BaseServiceImplTest.PROFILE_TYPE_2, molecularProfiles);
+   
+        Mockito.when(molecularProfileUtil.categorizeMolecularProfilesByStableIdSuffixes(molecularProfiles))
+            .thenReturn(molecularProfileMap);
+        
+        List<Gene> genes = new ArrayList<>();
+        Gene gene1 = new Gene();
+        gene1.setEntrezGeneId(BaseServiceImplTest.ENTREZ_GENE_ID_1);
+        gene1.setHugoGeneSymbol(BaseServiceImplTest.HUGO_GENE_SYMBOL_1);
+        Gene gene2 = new Gene();
+        gene2.setEntrezGeneId(BaseServiceImplTest.ENTREZ_GENE_ID_2);
+        gene2.setHugoGeneSymbol(BaseServiceImplTest.HUGO_GENE_SYMBOL_2);
+        genes.add(gene1);
+        genes.add(gene2);
+
+        Mockito.when(geneService.fetchGenes(anyList(), anyString(), anyString()))
+                .thenReturn(genes);
+
+        List<GeneMolecularData> geneMolecularData = new ArrayList<>();
+        GeneMolecularData geneMolecularData1 = new GeneMolecularData();
+        geneMolecularData1.setValue("-2");
+        GeneMolecularData geneMolecularData2 = new GeneMolecularData();
+        geneMolecularData2.setValue("-2");
+        GeneMolecularData geneMolecularData3 = new GeneMolecularData();
+        geneMolecularData3.setValue("2");
+        GeneMolecularData geneMolecularData4 = new GeneMolecularData();
+        geneMolecularData4.setValue("2");
+        geneMolecularData.add(geneMolecularData1);
+        geneMolecularData.add(geneMolecularData2);
+        geneMolecularData.add(geneMolecularData3);
+        geneMolecularData.add(geneMolecularData4);
+        
+        Mockito.when(molecularDataService.getMolecularDataInMultipleMolecularProfiles(
+            anyList(),
+            anyList(),
+            anyList(),
+            anyString()))
+            .thenReturn(geneMolecularData);
+        
+        List<GenomicDataCountItem> result = studyViewService.getCNAAlterationCountsByGeneSpecific(studyIds, sampleIds, genomicDataFilters);
+        Assert.assertEquals(2L, result.size());
```

Comment:
Just curious why use `2L` here?

Reply:
count 2(Long)

Reply:
@kalletlak It's just `2` is a small number and the default result.size() type is int, so it looks strange to specify `2L` here.